### PR TITLE
Allow TS type augmentation for Fastify.prototype.use

### PR DESCRIPTION
**Pull request by joelmukuthu**
This change allows use of `Fastify.prototype.use` as is or with custom type augmentations. 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: types/instance.d.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export interface FastifyInstance<
    * This method is now deprecated and will throw a `FST_ERR_MISSING_MIDDLEWARE` error.
```
Comment (mcollina) [inline types/instance.d.ts:57]: This comment is false. It's not deprecated and you'd need to use https://github.com/fastify/middie or https://github.com/fastify/fastiy-express.

Comment (joelmukuthu) [reply types/instance.d.ts:57]: Okidok, I'll update it. Then would you prefer I describe the arguments in full (instead of `unknown[]`)?

Comment (mcollina) [reply types/instance.d.ts:57]: Yes please!

File: types/instance.d.ts [hunk 2]
```diff
@@ -53,11 +53,28 @@ export interface FastifyInstance<
...
+   * https://github.com/fastify/fastiy-express.
    */
-  use: void;
+  use(
+    middleware: (
+      req: RawRequest,
+      res: RawReply,
+      next: (err?: Error | unknown) => void
+    ) => void
+  ): void;
```
Comment (delvedor) [inline types/instance.d.ts:69]: Calling `use` returns the Fastify instance in both `fastify-express` and `middie` :)

Comment (joelmukuthu) [reply types/instance.d.ts:69]: Wonderful, thanks! :tada:

File: types/instance.d.ts [hunk 3]
```diff
@@ -53,11 +53,28 @@ export interface FastifyInstance<
...
+
   /**
-   * This method is now deprecated and will throw a `FST_ERR_MISSING_MIDDLEWARE` error.
-   * Visit fastify.io/docs/latest/Middleware/ for more info.
+   * Mounts an Express-style middleware. However, support for Express-style
+   * middlewares should first be enabled via either
+   * https://github.com/fastify/middie or
+   * https://github.com/fastify/fastiy-express.
    */
-  use: void;
```
Comment (Ethan-Arrowood) [inline types/instance.d.ts:60]: to @delvedor point if this type is only available when using those plugins then this type needs to be augmented there and not here on master.

File: types/instance.d.ts (hunk 1 above)
Comment (Ethan-Arrowood) [reply types/instance.d.ts]: https://github.com/fastify/fastify/blob/7ddbef9d0f2234b506cb1a3849c027f72f1c9c04/fastify.js#L392

Is this comment still false?

Comment (joelmukuthu) [reply types/instance.d.ts]: I left out the `@throws FST_ERR_MISSING_MIDDLEWARE` to reduce the risk of that bit of documentation getting outdated. If anything I'd rather remove the comment entirely or have it point to the docs instead.

**Comment by joelmukuthu** [comment]
@mcollina are you okay with outsourcing this to the plugins instead?

File: types/instance.d.ts (hunk 1 above)
Comment (joelmukuthu) [reply types/instance.d.ts]: But to be clear, @Ethan-Arrowood, would you rather I add back the note about the error?

Comment (joelmukuthu) [reply types/instance.d.ts]: Doh. Nevermind, I think I get it now.

**Comment by Ethan-Arrowood** [comment]
I like the JSDoc change

**Comment by mcollina** [comment]
> @mcollina are you okay with outsourcing this to the plugins instead?

I'm ok with it!

**Pull request merged**